### PR TITLE
imp(apps): added 'WithICS4Wrapper' function to keepers

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

This is first one of the upcoming PRs requested by @colin-axner to reduce the diff when reviewing https://github.com/cosmos/ibc-go/pull/3939

This function allows to set the middleware after the keeper's creation.

closes: #XXXX

### Commit Message / Changelog Entry

```text
imp(apps): added 'WithICS4Wrapper' function to keepers
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] ~Linked to Github issue with discussion and accepted design OR link to spec that describes this work.~
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] ~Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).~
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by srdtrk** [comment]
In that case, should I add it to icahost too @colin-axner?

**Comment by colin-axner** [comment]
> In that case, should I add it to icahost too @colin-axner?

Good question. It's not necessary, but it might be nice to do for consistency sake. It's useful to define the ics4wrapper after generating all the ibc modules. So I would be in favour of doing so, but I can go either way. Happy to wait to hear if others have an opinion next week

File: modules/apps/29-fee/keeper/keeper_test.go [hunk 1]
```diff
@@ -297,3 +299,24 @@ func (suite *KeeperTestSuite) TestGetAllCounterpartyPayees() {
...
+
+	// test if the ics4 wrapper is the channel keeper initially
+	ics4Wrapper := suite.chainA.GetSimApp().IBCFeeKeeper.GetICS4Wrapper()
+
+	_, isChannelKeeper := ics4Wrapper.(channelkeeper.Keeper)
+	suite.Require().True(isChannelKeeper)
+	_, isFeeKeeper := ics4Wrapper.(keeper.Keeper)
+	suite.Require().False(isFeeKeeper)
+
+	// set the ics4 wrapper to itself (don't do this in production)
```
Comment (crodriguezvega) [inline modules/apps/29-fee/keeper/keeper_test.go:314]: 😄

**Review by crodriguezvega — APPROVED** [review]

File: modules/apps/27-interchain-accounts/controller/keeper/keeper.go [hunk 2]
```diff
@@ -66,6 +66,13 @@ func NewKeeper(
 	}
 }
 
+// WithICS4Wrapper sets the ICS4Wrapper. This function may be used after
+// the keepers creation to set the middleware which is above this module
+// in the IBC application stack.
```
[the remark this answers is not in the document]
Comment (srdtrk) [reply modules/apps/27-interchain-accounts/controller/keeper/keeper.go:71]: I can't think of other circumstances :sweat_smile:

**Review by damiannolan — APPROVED** [review]

**Pull request merged**